### PR TITLE
added minio storage, docker-compose, glasswall pdf scrape

### PR DESCRIPTION
**Pull request by belwase**

### CODE REVIEW

File: upwork-devs/ashish-belwase/virusshare_poc/src/minio_service.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
+import os
+import logging
+import boto3
+import requests
+import time
+from botocore.client import Config
+from botocore.exceptions import ClientError
+
+logger = logging.getLogger("GW:minio")
```
Comment (susilupwork) [inline upwork-devs/ashish-belwase/virusshare_poc/src/minio_service.py:9]: @pranaysahith  Very good. At least seen logging now.

File: upwork-devs/ashish-belwase/virusshare_poc/src/minio_service.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    def download_files(self, bucket_name, num_files):
+
+        try:
+            file_path = "./"
+            logger.debug("Check if the Bucket {} exists".format(bucket_name))
+            if self.s3.Bucket(bucket_name) not in self.s3.buckets.all():
+                raise Exception(f"{bucket_name} bucket does not exist")
+            bucket = self.s3.Bucket(bucket_name)
+            files_list = []
+            saved_files = 0
```
Comment (susilupwork) [inline upwork-devs/ashish-belwase/virusshare_poc/src/minio_service.py:32]: S3 code should be in separate file.

File: upwork-devs/ashish-belwase/virusshare_poc/src/minio_service.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+            logger.error(e)
+
+    def upload_to_minio(self, file_path, bucket_name, filename):
+
+        try:
+            logger.debug("Checking if the Bucket to upload files exists or not.")
+            if (self.s3.Bucket(bucket_name) in self.s3.buckets.all()) == False:
+                logger.info("Bucket not Found. Creating Bucket.")
+                self.s3.create_bucket(Bucket=bucket_name)
+            logger.debug(
```
Comment (susilupwork) [inline upwork-devs/ashish-belwase/virusshare_poc/src/minio_service.py:59]: Please make this INFO logging. Any write operation should be INFO with relevant variable

File: upwork-devs/ashish-belwase/virusshare_poc/src/minio_service.py (hunk 2 above)
Comment (pranaysahith) [reply upwork-devs/ashish-belwase/virusshare_poc/src/minio_service.py:32]: we are using boto3 sdk to interact with minio server. 
we will have a separate file/class when working with actual aws s3

File: upwork-devs/ashish-belwase/virusshare_poc/src/minio_service.py (hunk 3 above)
Comment (pranaysahith) [reply upwork-devs/ashish-belwase/virusshare_poc/src/minio_service.py]: updated

**Review by susilupwork — APPROVED** [review]
- Minio service should have loose coupling with AWS S3 SDK. , Create S3 Adapter. 
- All write operation/ Network operation should have INFO logging, covering all critical parameters, like variables, users etc. 
- All minio storage service usage should have analytics data captured in storing and retrieval. Check with  developers channel about minio analytics best practices. 
-   As this is a opensource project, Please keep your Secret key, password etc. in environment variable and put in README example variables. 
- BaseScrapper part is ignored because that is only for proof of concept and today's delivery.

**Review by DinisCruz — APPROVED** [review]

**Pull request merged**